tuna/miopen/driver/convolution.py

```diff
@@ -56,37 +63,37 @@ def __init__(self, line=None, cmd=None, db_obj=None, kwargs=None):
         'in_channels', 'out_channels', 'direction', 'cmd'
     ])
 
-    self.batchsize = None
-    self.spatial_dim = None
-    self.pad_h = None
-    self.pad_w = None
-    self.pad_d = None
-    self.conv_stride_h = None
-    self.conv_stride_w = None
-    self.conv_stride_d = None
-    self.dilation_h = None
-    self.dilation_w = None
-    self.dilation_d = None
-    self.group_count = None
-    self.mode = None
-    self.pad_mode = None
-    self.trans_output_pad_h = None
-    self.trans_output_pad_w = None
-    self.trans_output_pad_d = None
-    self.out_layout = None
-    self.in_layout = None
-    self.fil_layout = None
-    self.in_d = None
-    self.in_h = None
-    self.in_w = None
-    self.fil_d = None
-    self.fil_h = None
-    self.fil_w = None
-    self.in_channels = None
-    self.out_channels = None
-    self.num_dims = None
-    self.direction = None
-    self._cmd = 'conv'
+    self.batchsize: Optional[str] = None
+    self.spatial_dim: Optional[str] = None
+    self.pad_h: Optional[str] = None
+    self.pad_w: Optional[str] = None
+    self.pad_d: Optional[str] = None
+    self.conv_stride_h: Optional[str] = None
+    self.conv_stride_w: Optional[str] = None
+    self.conv_stride_d: Optional[str] = None
+    self.dilation_h: Optional[str] = None
+    self.dilation_w: Optional[str] = None
+    self.dilation_d: Optional[str] = None
+    self.group_count: Optional[str] = None
+    self.mode: Optional[str] = None
+    self.pad_mode: Optional[str] = None
+    self.trans_output_pad_h: Optional[str] = None
+    self.trans_output_pad_w: Optional[str] = None
+    self.trans_output_pad_d: Optional[str] = None
+    self.out_layout: Optional[str] = None
+    self.in_layout: Optional[str] = None
+    self.fil_layout: Optional[str] = None
+    self.in_d: Optional[str] = None
+    self.in_h: Optional[str] = None
+    self.in_w: Optional[str] = None
+    self.fil_d: Optional[str] = None
+    self.fil_h: Optional[str] = None
+    self.fil_w: Optional[str] = None
+    self.in_channels: Optional[str] = None
+    self.out_channels: Optional[str] = None
+    self.num_dims: Optional[int] = None
+    self.direction: Optional[str] = None
+    self._cmd: Optional[str] = 'conv'
```

Comment:
dont think this is optional, should just be str

Reply:
There's 3 ways these values could be populated. 
It could be the kwargs from the constructor here. - and so set here immediately
It could be from parsing the driver line. - set by super constructor
It could be from parsing the fdb_key from a db file. - set by super constructor

Reply:
Potential solution is to populate these fields with the defaults for conv2d in metadata.

Reply:
data set is initialized to CONV_2D_DEFAULTS table from metadata.py